addons/html_editor/static/src/others/embedded_components/plugins/toggle_plugin/toggle_plugin.js

```diff
@@ -0,0 +1,447 @@
+            const [rightNode, rightOffset] = rightPos(container);
+            this.dependencies.selection.setSelection({
+                anchorNode: rightNode,
+                anchorOffset: rightOffset,
+                focusNode: rightNode,
+                focusOffset: rightOffset,
+            });
+            const fragment = this.document.createDocumentFragment();
+            fragment.replaceChildren(...contentToInsert);
+            this.dependencies.dom.insert(fragment);
+        }
+        container.replaceWith(newParagraph);
+        this.dependencies.history.addStep();
+        this.dependencies.selection.setCursorStart(newParagraph);
+        return true;
+    }
+
+    /**
+     * Handles the tab behavior. This means that when we are inside a toggle title and we have a toggle
+     * as previous sibling of the embedded component, the current toggle is indented inside the content of
+     * the previous one.
+     */
+    handleTab() {
+        const container = this.getToggleContainer();
+        if (container) {
+            // If selection is in a title.
+            const previousSibling = container.previousElementSibling;
+            if (previousSibling?.matches("[data-embedded='toggle']")) {
+                // If the previous element of the embedded component is also a toggle, we need to handle
+                // it.
+                const { restore: restoreSelection } =
+                    this.dependencies.selection.preserveSelection();
+                this.dependencies.selection.setCursorEnd(
+                    previousSibling.querySelector("[data-embedded-editable='content']")
+                );
+                previousSibling.firstChild.querySelector("i.fa-caret-right")?.click(); // open the toggle if needed
+                const fragment = this.document.createDocumentFragment();
+                fragment.replaceChildren(container);
+                const canReplace = previousSibling.querySelector(
+                    "[data-embedded-editable='content'] > *:only-child"
+                );
+                // If the 1st block of the previous content is empty we replace it with our toggle.
+                // Else we add it to the content.
+                if (isEmptyBlock(canReplace)) {
+                    canReplace.replaceWith(fragment);
+                } else {
+                    this.dependencies.dom.insert(fragment);
+                }
+                this.dependencies.history.addStep();
+                window.setTimeout(restoreSelection, "animationFrame"); // Used to handle caret displaying issues
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Handles the shift-tab behavior. This means that we need to outdent the toggle from each other.
+     * @returns
+     */
+    handleShiftTab() {
+        const container = this.getToggleContainer();
+        if (container) {
+            if (container.parentElement.closest("[data-embedded='toggle']")) {
```

Comment:
Use `closestElement` to limit the search to the current editable.
You could store the result in a variable to avoid searching twice